signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x1b6c1d6

Hi, I get this error using the latest version of Nuclei from MacBook Pro 2019

**Nuclei Engine 2.8.1 (latest)
Nuclei Templates 9.3.0 (latest)**

```
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x1b6c1d6]

goroutine 1481646 [running]:
github.com/projectdiscovery/nuclei/v2/pkg/input.(*Helper).convertInputToType(0xc0024b6028, {0xc00afe0d00, 0xc8}, 0x4, {0x0, 0x0})
	github.com/projectdiscovery/nuclei/v2/pkg/input/input.go:72 +0x96
github.com/projectdiscovery/nuclei/v2/pkg/input.(*Helper).Transform(0x1046973?, {0xc00afe0d00?, 0x8?}, 0x1304efd?)
	github.com/projectdiscovery/nuclei/v2/pkg/input/input.go:43 +0x7f
github.com/projectdiscovery/nuclei/v2/pkg/protocols/common/executer.(*Executer).Execute(0xc0021b0d60, 0xc0033c2940)
	github.com/projectdiscovery/nuclei/v2/pkg/protocols/common/executer/executer.go:75 +0x2c6
github.com/projectdiscovery/nuclei/v2/pkg/core.(*Engine).executeModelWithInput.func2.1(0x132a226?, 0x0?, 0xc0033c2800)
	github.com/projectdiscovery/nuclei/v2/pkg/core/execute.go:146 +0x182
created by github.com/projectdiscovery/nuclei/v2/pkg/core.(*En
	github.com/projectdiscovery/nuclei/v2/pkg/core/execute.go:131 +0x745

```

@tarunKoyalwar @vzamanillo @Ice3man543 from https://github.com/projectdiscovery/nuclei/issues/2996

```bash
$ echo 'https://~^(.*-pre)\.smth\.google\.com/' | nuclei -id tech-detect -silent

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x28 pc=0x10343eb48]

goroutine 59 [running]:
github.com/projectdiscovery/nuclei/v2/pkg/input.(*Helper).convertInputToType(0x1400052c028, {0x14002439f50, 0x26}, 0x4, {0x0, 0x0})
	github.com/projectdiscovery/nuclei/v2/pkg/input/input.go:72 +0x68
github.com/projectdiscovery/nuclei/v2/pkg/input.(*Helper).Transform(0x1400053a608?, {0x14002439f50?, 0x1400053a5a8?}, 0x1029abde8?)
	github.com/projectdiscovery/nuclei/v2/pkg/input/input.go:43 +0x88
github.com/projectdiscovery/nuclei/v2/pkg/protocols/common/executer.(*Executer).Execute(0x14001ef83e0, 0x14002449400)
	github.com/projectdiscovery/nuclei/v2/pkg/protocols/common/executer/executer.go:75 +0x298
github.com/projectdiscovery/nuclei/v2/pkg/core.(*Engine).executeModelWithInput.func2.1(0x0?, 0x0?, 0x14002449360)
	github.com/projectdiscovery/nuclei/v2/pkg/core/execute.go:146 +0x148
created by github.com/projectdiscovery/nuclei/v2/pkg/core.(*Engine).executeModelWithInput.func2
	github.com/projectdiscovery/nuclei/v2/pkg/core/execute.go:131 +0x618
```

--- a/v2/pkg/input/input.go
+++ b/v2/pkg/input/input.go
@@ -66,9 +66,15 @@ const (
 // Various formats are supported for inputs and their transformation
 func (h *Helper) convertInputToType(input string, inputType inputType, defaultPort string) string {
 	notURL := !strings.Contains(input, "://")
+	if strings.Contains(input, "/") {
+		// edge case if input is similar to github.com/projectdiscovery or
+		// example.com:80/admin
+		notURL = false
+		input = "http://" + input
+	}
 	parsed, _ := url.Parse(input)
 	var host, port string
-	if !notURL {
+	if !notURL && parsed != nil {
 		host, port, _ = net.SplitHostPort(parsed.Host)
 	} else {
 		host, port, _ = net.SplitHostPort(input)

--- a/v2/pkg/input/input.go
+++ b/v2/pkg/input/input.go
@@ -66,12 +66,6 @@ const (
 // Various formats are supported for inputs and their transformation
 func (h *Helper) convertInputToType(input string, inputType inputType, defaultPort string) string {
 	notURL := !strings.Contains(input, "://")
-	if strings.Contains(input, "/") {
-		// edge case if input is similar to github.com/projectdiscovery or
-		// example.com:80/admin
-		notURL = false
-		input = "http://" + input
-	}
 	parsed, _ := url.Parse(input)
 	var host, port string
 	if !notURL && parsed != nil {

